packages/zoe/src/contractSupport/priceAuthority.js

```diff
@@ -301,7 +301,7 @@ export function makeOnewayPriceAuthorityKit(opts) {
       return quote;
     },
     async quoteAtTime(deadline, amountIn, brandOut) {
-      assert.typeof(deadline, 'bigint');
+      assert.typeof(deadline, 'object'); // Timestamp
```

Comment:
fit(deadline, TIMESTAMP_SHAPE);

Reply:
@Chris-Hibbert I've got a patch for this:
```diff
     async quoteAtTime(deadline, amountIn, brandOut) {
+      mustMatch(deadline, TimestampShape);
       AmountMath.coerce(actualBrandIn, amountIn);
       assertBrands(amountIn.brand, brandOut);
```

but I'm worried that it will impose a change on the clients of Zoe that they may not be prepared to handle yet. We decided to reduce the goals of this PR: have the timerService *return* branded timestamps, but accept either. So a lot of the changes are to ensure that when Zoe receives a timestamp, it can deal with it, but we want to keep everything tolerant of code that' still sending plain BigInts into time stuff, including zoe. If we can get that into the Vaults release, then contracts can begin to write code that matches our long-term goals (always use brands), because both "send bigints, accept brands" and "send brands, accept brands" will work.

Do you think I should apply this `mustMatch` change?

Reply:
For this one, we decided to use `deadline = coerceToTimestampRecord(deadline, timerBrand)`, which will accept any of `number | bigint | TimestampRecord (with correct brand)`, but reject TimestampRecords with the wrong brand.

This will succeed or fail in exactly the same situations as the current PR's lack of any assertions, but the failure will happen as the first line of `quoteAtTime`, instead of happening inside the timer service a moment later.

Reply:
Actually, this is turning out to be awkward to implement. The enclosing function (`makeOnewayPriceAuthorityKit`) is not `async`, which means I can't do `timerBrand = await E(timerService).getTimerBrand()` outside of each call to `quoteAtTime` call (which would add some overhead that I'd like to avoid). The next earliest place to fetch the brand is in the four callers to `makeOnewayPriceAuthorityKit()`:
* zoe/src/contracts/priceAggregator.js
* inter-protocol/src/price/fluxAggregatorKit.js
* zoe/tools/manualPriceAuthority.js
* zoe/tools/scriptedPriceAuthority.js

The first two are async, and not a problem. The `zoe/tools/` ones are sync, but it probably also use a local (manual)TimerService, so in theory I can just do `timerBrand = timerService.getTimerBrand()`, but the typechecker complains because it thinks that `timerService` is an ERef.

I think I'm not going to fix this one, and rely upon TimerService complaining about a mismatched brand, instead of trying to catch it slightly earlier.

Reply:
Since the two sync calls are from test tools, it would be fine to `// @ts-expect-error ` them.